activerecord/test/cases/assertions/query_assertions_test.rb

```diff
@@ -21,6 +21,13 @@ def test_assert_queries
           assert_queries(0) { Post.first }
         }
         assert_match(/1 instead of 0 queries/, error.message)
+
+        error = assert_raises(Minitest::Assertion) {
+          assert_queries(1, matcher: /WHERE "posts"."id" = \? LIMIT \?/) do
+            Post.where(id: 1).first
+          end
+        }
+        assert_match(/0 instead of 1 queries/, error.message)
```

Comment:
Thanks for adding this 🙏
Could you move this to a separate test case method?

Reply:
Thank you! It makes sense to me. I moved the code with 7a268ecd1bd7296e0fea0a7663f812374da05087.

By the way, I couldn't run the test on my local machine. It's weird...

<img width="691" alt="image" src="https://github.com/rails/rails/assets/13130705/745f022d-0a21-4b64-8e17-c9f426a9d716">